ui/app/models/job.js

```diff
@@ -366,4 +366,17 @@ export default class Job extends Model {
       return this.variables?.findBy('path', `nomad/jobs/${this.plainId}`);
     }
   }
+
+  // TODO: This async fetcher seems like a better fit for most of our use-cases than the above getter (which cannot do async/await)
+  async getPathLinkedVariable() {
+    await this.variables;
+    if (this.parent.get('id')) {
+      return await this.variables?.findBy(
+        'path',
+        `nomad/jobs/${JSON.parse(this.parent.get('id'))[0]}`
+      );
```

Comment:
If `this.variables` here is a RecordCollection (resolved relationship, however you want to call it), then isn't `findBy` here a synchronous array operation? If so, the `await` is doing anything.

Reply:
Guaranteeing that the variables are loaded without pushing that logic onto components or controllers is nice though.